### ISSUE TITLE
Private packages should not be listed in publish success messages

### ISSUE THREAD

**Issue by superkhau** [lerna/lerna#810]
Running `lerna publish` prompts for individual package versions for packages marked with `private: true`. 

```
lerna publish
? Select a new version for @org/package-a
? Enter a custom version 1.0.0-alpha.1 # Forced to manually enter a dummy version

Changes:
 - @org/package-a: 1.0.0-alpha.1 => 1.0.0-alpha.1 (private) # This is confusing
 - @org/package-b: 1.0.0-alpha.1 => 1.0.0-alpha.1
 - ...

? Are you sure you want to publish the above changes? Yes
lerna info publish Publishing packages to npm...
lerna info published @package-b
lerna info git Pushing tags...
Successfully published:
 - @org/package-a: 1.0.0-alpha.1 => 1.0.0-alpha.1 # This is confusing
 - @org/package-b: 1.0.0-alpha.1 => 1.0.0-alpha.1
lerna success publish finished
```

While it doesn't actually publish the private packages to NPM, the console output still shows it as published successfully.

**Comment by evocateur** [comment]
If we don't prompt for them, what version are we supposed to use? Private packages are intended to still be versioned, just never published. I agree the messaging is misleading, and would appreciate a pull request separating them appropriately.

**Comment by superkhau** [comment]
>If we don't prompt for them, what version are we supposed to use? Private packages are intended to still be versioned, just never published.

You're right. Maybe we should add an option to skip versioning instead?

```
Package X@1.0.0-alpha is marked as private, skip versioning?
```

That way, the CLI tool can skip the other weird prompts below.

As for sending a PR, my schedule is pretty full and won't be available again till beginning of June. I'll try to send then if someone doesn't get around to it before me. ;)

**Comment by evocateur** [comment]
I understand :)

**Comment by Reinmar** [comment]
There's an even bigger problem with private packages:

![image](https://cloud.githubusercontent.com/assets/156149/26108166/fce83184-3a4b-11e7-8ede-40acf3d3b562.png)

Lerna tries to release version `null` which even works for the first time, but fails for the second :D.

**Comment by evocateur** [comment]
Oh my, that's ...unfortunate. In independent mode, it sounds like we need to validate that every package has a valid version? I think the null is a raw result from semver.valid(), which really means our prompt validation needs some cleanup.

**Comment by donaldpipowitch** [comment]
> Private packages are intended to still be versioned, just never published.

Could we opt-out from this behaviour somehow?

**Comment by evocateur** [comment]
A version is a fundamental part of an npm package, so no, I don’t see how you opt-out of versioning a private package (short of no longer treating it as a package, then go to town I guess?).

**Comment by donaldpipowitch** [comment]
Well it "is" versioned in the sense of "it has a version". But I don't ever want or need to update the version, because they are never consumed by anyone.

**Comment by cezaraugusto** [comment]
I guess this was fixed by #846, If https://github.com/lerna/lerna/issues/810#issuecomment-301781565 is a concern it could be logged separately in a different issue
